tabular/src/autogluon/tabular/learner/default_learner.py

```diff
@@ -72,6 +72,8 @@ def _fit(self, X: DataFrame, X_val: DataFrame = None, X_unlabeled: DataFrame = N
         time_preprocessing_start = time.time()
         logger.log(20, 'Preprocessing data ...')
         self._pre_X_rows = len(X)
+        self._original_features = list(X)
+        self._original_features.remove(self.label)
```

Comment:
Not quite correct, this could contain additional features still such as sample weight and `self.groups` which wouldn't be present at test time. A better option is to record original features in the feature generator object, since this is the stage at which all of those extra columns have been removed and only the true features are present.

Reply:
Another option: Add self._original_features right before the call to fit feature generator, this will work identically to current implementation, but should be after the other columns are removed and only the real features are present. This might be a bit better than making the logic part of FeatureGenerator.